src/parsers/texttracks/webvtt/__tests__/utils.test.ts

```diff
@@ -0,0 +1,402 @@
+      expect(isStartOfCueBlock(webvtt3, 16)).to.equal(true);
+      expect(isStartOfCueBlock(webvtt3, 21)).to.equal(true);
+      expect(isStartOfCueBlock(webvtt4, 2)).to.equal(true);
+      expect(isStartOfCueBlock(webvtt5, 6)).to.equal(true);
+    });
+  });
+
+  describe("isStartOfNoteBlock", () => {
+    it("should return true if called on a `NOTE` line followed by timings", () => {
+      expect(isStartOfNoteBlock(webvtt2, 2)).to.equal(false);
+    });
+
+    it("should return true if called on a `NOTE` line not followed by timings", () => {
+      expect(isStartOfNoteBlock(webvtt6, 5)).to.equal(true);
+      expect(isStartOfNoteBlock(webvtt3, 18)).to.equal(true); // This is actually bad
+    });
+
+    it("should return true if called on line containing `NOTE` and spaces", () => {
+      expect(isStartOfNoteBlock(["NOTE    "], 0)).to.equal(true);
+      expect(isStartOfNoteBlock(["", "NOTE ", "TOTO"], 1)).to.equal(true);
+    });
+
+    /* tslint:disable max-line-length */
+    it("should return true if called on line containing `NOTE` and spaces and text", () => {
+    /* tslint:enable max-line-length */
+      expect(isStartOfNoteBlock(webvtt1, 9)).to.equal(true);
+      expect(isStartOfNoteBlock(webvtt1, 19)).to.equal(true);
+    });
+
+    /* tslint:disable max-line-length */
+    it("should return false if called on a line containing `NOTE` and text attached", () => {
+    /* tslint:enable max-line-length */
+      expect(isStartOfNoteBlock(["NOTEdsj f"], 0)).to.equal(false);
+      expect(isStartOfNoteBlock(["aaa", "NOTEoej ewj ", "aaa"], 1)).to.equal(false);
+    });
+
+    it("should return false if called on a region block", () => {
+      expect(isStartOfNoteBlock(["REGION SOMETHING"], 0)).to.equal(false);
+    });
+
+    it("should return false if called on a style block", () => {
+      expect(isStartOfNoteBlock(["STYLE SOMETHING"], 0)).to.equal(false);
+      expect(isStartOfCueBlock(webvtt1, 2)).to.equal(false);
+      expect(isStartOfCueBlock(webvtt1, 11)).to.equal(false);
+    });
+
+    it("should return false if called on an empty line", () => {
+      expect(isStartOfNoteBlock(["", "NOTE"], 0)).to.equal(false);
+      expect(isStartOfCueBlock(webvtt1, 18)).to.equal(false);
+      expect(isStartOfCueBlock(webvtt3, 1)).to.equal(false);
+    });
+  });
+
+  describe("isStartOfRegionBlock", () => {
+    it("should return true if called on a `REGION` line", () => {
+      expect(isStartOfRegionBlock(["REGION"], 0)).to.equal(true);
+    });
+
+    it("should return true if called on line containing `REGION` and spaces", () => {
+      expect(isStartOfRegionBlock(["REGION "], 0)).to.equal(true);
+      expect(isStartOfRegionBlock(["REGION  "], 0)).to.equal(true);
+      expect(isStartOfRegionBlock(["REGION         "], 0)).to.equal(true);
+    });
+
+    /* tslint:disable max-line-length */
+    it("should return true if called on line containing `REGION` and spaces and text", () => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Update or refactor this function so that its implementation doesn't duplicate the one on line 257. ([typescript:S4144](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=229&issues=AWOsNmQvmXhJxGlVaS-j&open=AWOsNmQvmXhJxGlVaS-j)

[ISSUE_KEY]: AWOsNmQvmXhJxGlVaS-j